mediator/src/routes/mod.rs

```diff
@@ -0,0 +1,215 @@
+        //         &unpacked
+        //             .sender_verkey
+        //             .ok_or("Can't register anon aries peer")?,
+        //         did_doc,
+        //         &unpacked.recipient_verkey,
+        //         xum_test_server::didcomm_types::mediator_coord_structs::MediateGrantData{
+        // endpoint: service_endpoint.into(), routing_keys},     );
+        //     todo!()
+        } else {
+            let (account_name, our_signing_key, their_diddoc) =
+                agent.auth_and_get_details(&unpacked.sender_verkey).await?;
+            let auth_pubkey = unpacked
+                .sender_verkey
+                .expect("Sender key authenticated above, so it must be present..");
+            info!("Processing message for {:?}", account_name);
+            match aries_message {
+                GeneralAriesMessage::AriesVCXSupported(aries_message) => {
+                    Err(unhandled_aries(aries_message))?
+                }
+                GeneralAriesMessage::XumCoord(coord_message) => {
+                    let coord_response =
+                        handle_mediation_coord(&agent, coord_message, &auth_pubkey).await?;
+                    let aries_response =
+                        serde_json::to_vec(&coord_response).map_err(string_from_std_error)?;
+                    agent
+                        .pack_didcomm(&aries_response, &our_signing_key, &their_diddoc)
+                        .await?
+                }
+                GeneralAriesMessage::XumPickup(pickup_message) => {
+                    handle_pickup_protocol(agent, pickup_message).await?;
+                    todo!();
+                }
+            }
+        };
+    let EncryptionEnvelope(packed_message_bytes) = packed_response;
+    let packed_json = serde_json::from_slice(&packed_message_bytes[..]).unwrap();
+    Ok(Json(packed_json))
+}
+pub async fn oob_invite_qr(
+    headers: HeaderMap,
+    State(agent): State<ArcAgent<impl BaseWallet + 'static, impl MediatorPersistence>>,
+) -> Response {
+    let Json(oob_json) = oob_invite_json(State(agent)).await;
+    let preferred_mimetype = headers
+        .get(ACCEPT)
+        .map(|s| s.to_str().unwrap_or_default())
+        .unwrap_or_default();
+    match preferred_mimetype {
+        "application/json" => Json(oob_json).into_response(),
+        _ => {
+            let oob_string = serde_json::to_string_pretty(&oob_json).unwrap();
+            let qr = fast_qr::QRBuilder::new(oob_string.clone()).build().unwrap();
+            let oob_qr_svg = fast_qr::convert::svg::SvgBuilder::default().to_str(&qr);
+            Html(format!(
+                "<style>
+                        svg {{
+                            width: 50%;
+                            height: 50%;
+                        }}
+                    </style>
+                    {oob_qr_svg} <br>
+                    <pre>{oob_string}</pre>"
+            ))
+            .into_response()
+        }
+    }
```

Comment:
this for example should stay in this layer - it's only API interface specific thing that you want to provide rendered QR codes. 
But `oob_invite_json` should be in service layer